src/js/actions/layers.js

```diff
@@ -892,7 +894,7 @@ define(function (require, exports) {
             .bind(this)
             .then(function (event) {
                 var layerID = event.layerID;
-                return this.transfer(addLayers, document, layerID, true, true);
+                return this.transfer(addLayers, document, layerID, true, layer.id);
```

Comment:
addLayersCommand's JSDoc suggests that replace takes an index, yet you're passing the layer ID here.